subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2025,24 +2025,21 @@ static void le_set_cig_parameters(struct net_buf *buf, struct net_buf **evt)
 
 	rp = hci_cmd_complete(evt, sizeof(*rp) + cis_count * sizeof(uint16_t));
 	rp->cig_id = cig_id;
-	rp->num_handles = cis_count;
 
 	/* Only apply parameters if all went well */
 	if (!status) {
-		status = ll_cig_parameters_commit(cig_id);
+		uint16_t handles[CONFIG_BT_CTLR_CONN_ISO_STREAMS_PER_GROUP];
 
-		if (status == BT_HCI_ERR_SUCCESS) {
-			struct ll_conn_iso_group *cig;
-			uint16_t handle;
-
-			cig = ll_conn_iso_group_get_by_id(cig_id);
-			handle = UINT16_MAX;
+		rp->num_handles = cis_count;
+		status = ll_cig_parameters_commit(cig_id, handles);
 
+		if (status == BT_HCI_ERR_SUCCESS) {
 			for (uint8_t i = 0; i < cis_count; i++) {
-				(void)ll_conn_iso_stream_get_by_group(cig, &handle);
-				rp->handle[i] = sys_cpu_to_le16(handle);
+				rp->handle[i] = sys_cpu_to_le16(handles[i]);
 			}
```

Comment:
In case that `status != BT_HCI_ERR_SUCCESS`, are the `rp->handle[i]` value uninitialized? Or are they set to 0?

Reply:
Right, it's only if parameter checking fails. I'll add it to the case where commit fails